compiler/noirc_frontend/src/tests/metaprogramming/skip_interpreter_on_fail.rs

```diff
@@ -0,0 +1,373 @@
+//! Tests for skipping the comptime interpreter when there are elaboration errors for
+//! comptime blocks, functions, or let statements.
+
+use crate::tests::check_errors;
+
+#[test]
+fn evaluate_separate_comptime_block_with_preceding_failure() {
+    // Guarantee that we run a comptime block even if a preceding comptime
+    // block contains errors.
+    let src = "
+    fn main() {
+        comptime {
+            let i: u8 = 1;
+            let _ = -i;
+                    ^^ Cannot apply unary operator `-` to type `u8`
+        }
+
+        comptime {
+            let x: i8 = 5 + 10;
+            assert_eq(x, 15);
+        }
+    }
+    ";
+    check_errors(src);
+}
+
+#[test]
+fn do_not_evaluate_comptime_block_with_failure() {
+    let src = "
+    fn main() {
+        comptime {
+            let x: i8 = 5 + 20;
+            // We would expect this to trigger an interpreter error if it were not
+            // for the type error below.
+            assert_eq(x, 15);
+            let i: u8 = 1;
+            let _ = -i;
+                    ^^ Cannot apply unary operator `-` to type `u8`
+        }
+    }
+    ";
+    check_errors(src);
+}
+
+#[test]
+fn do_not_evaluate_comptime_block_with_preceding_failure() {
+    let src = "
+    fn main() {
+        comptime {
+            let i: u8 = 1;
+            let _ = -i;
+                    ^^ Cannot apply unary operator `-` to type `u8`
+            let x: i8 = 5 + 20;
+            // We would expect this to trigger an interpreter error if it were not
+            // for the type error abbove.
+            assert_eq(x, 15);
+        }
+    }
+    ";
+    check_errors(src);
+}
+
+#[test]
+fn failing_comptime_function_not_run() {
+    let src = "
+    comptime mut global FLAG: bool = false; 
+    
+    fn main() {
+        comptime {
+            bad(); 
+            // We expect the `FLAG` to remain `false`
+            assert_eq(FLAG, false);
```

Comment:
Here we could check `assert_eq(FLAG, true)` and not expect an error either, assuming this block doesn't run at comptime because `bad` has errors.

Reply:
Good idea I did that